compiler/x/codegen/CompareAnalyser.cpp

```diff
@@ -730,8 +730,7 @@ void TR_X86CompareAnalyser::longOrderedCompareAndBranchAnalyser(TR::Node       *
    cg()->decReferenceCount(firstChild);
    cg()->decReferenceCount(secondChild);
 
-   if (deps)
-      deps->setMayNeedToPopFPRegisters(true);
+   deps->setMayNeedToPopFPRegisters(true);
```

Comment:
It is checked against nullptr in some cases and is just directly used in other cases. I removed this check because `deps` was just used (see line 717).